go/plugins/googleai/googleai.go

```diff
@@ -314,26 +315,46 @@ func generate(
 	return r, nil
 }
 
+func mapToStruct(m map[string]interface{}, v interface{}) error {
+	jsonData, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(jsonData, v)
+}
+
+// applyGenerationConfig applies the common generation configuration to the model
+// todo: support Gemini-specific configuration
+func applyGenerationConfig(gm *genai.GenerativeModel, c *ai.GenerationCommonConfig) {
+	if c.MaxOutputTokens != 0 {
+		gm.SetMaxOutputTokens(int32(c.MaxOutputTokens))
+	}
+	if len(c.StopSequences) > 0 {
+		gm.StopSequences = c.StopSequences
+	}
+	if c.Temperature != 0 {
+		gm.SetTemperature(float32(c.Temperature))
+	}
+	if c.TopK != 0 {
+		gm.SetTopK(int32(c.TopK))
+	}
+	if c.TopP != 0 {
+		gm.SetTopP(float32(c.TopP))
+	}
+}
+
 func newModel(client *genai.Client, model string, input *ai.ModelRequest) (*genai.GenerativeModel, error) {
-	gm := client.GenerativeModel(model)
-	gm.SetCandidateCount(1)
-	if c, ok := input.Config.(*ai.GenerationCommonConfig); ok && c != nil {
-		if c.MaxOutputTokens != 0 {
-			gm.SetMaxOutputTokens(int32(c.MaxOutputTokens))
-		}
-		if len(c.StopSequences) > 0 {
-			gm.StopSequences = c.StopSequences
-		}
-		if c.Temperature != 0 {
-			gm.SetTemperature(float32(c.Temperature))
-		}
-		if c.TopK != 0 {
-			gm.SetTopK(int32(c.TopK))
-		}
-		if c.TopP != 0 {
-			gm.SetTopP(float32(c.TopP))
-		}
+	var c ai.GenerationCommonConfig
+	if err := mapToStruct(input.Config.(map[string]interface{}), &c); err != nil {
```

Comment:
It's not always a map though. It only comes in as a map if the request is from Dev UI. If you run a flow with a Generate call that has a config set, it will behave correctly. So this change is now breaking that I'm pretty sure.